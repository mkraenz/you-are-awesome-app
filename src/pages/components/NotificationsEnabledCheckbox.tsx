import { DateTime } from "luxon";
import React, { Component } from "react";
import { CheckBox } from "react-native-elements";
import { connect } from "react-redux";
import { IReduxState } from "../../redux/IReduxState";
import { styles } from "../Styles";

interface Props {
    onPress: () => void;
    notificationsEnabled: boolean;
    scheduledTime: Date;
}

class NotificationsEnabledCheckbox extends Component<Props> {
    public render() {
        return (
            <CheckBox
                containerStyle={[styles.button, { borderWidth: 0 }]}
                textStyle={{ flex: 1, color: "white" }}
                checkedColor="white"
                uncheckedColor="white"
                center
                title={this.getTitle()}
                iconRight
                checked={this.props.notificationsEnabled}
                onPress={this.props.onPress}
            />
        );
    }

    private getTitle() {
        const base = "Enable daily notifications";
        const addition = `\nNext at ${DateTime.fromJSDate(
            this.props.scheduledTime
        )
            .toLocaleString(DateTime.DATETIME_MED)
            .slice(-8)
            .trim()}`;
        return this.props.notificationsEnabled ? `${base}${addition}` : base;
    }
}

const mapStateToProps = (
    state: Pick<IReduxState, "settings">
): Pick<Props, "notificationsEnabled" | "scheduledTime"> => ({
    notificationsEnabled: state.settings.notificationsEnabled,
    scheduledTime: state.settings.scheduledTime,
});

export default connect(mapStateToProps)(NotificationsEnabledCheckbox);
