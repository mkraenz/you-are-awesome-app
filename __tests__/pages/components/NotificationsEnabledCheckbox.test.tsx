import React from "react";
import "react-native";
import { Provider } from "react-redux";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import createMockStore from "redux-mock-store";
import NotificationsEnabledCheckbox from "../../../src/pages/components/NotificationsEnabledCheckbox";

it("renders correctly for disabled", () => {
    const scheduledTime = new Date("2029-09-06T15:00:00.000Z");
    const store = createMockStore([])({
        settings: {
            notificationsEnabled: false,
            scheduledTime,
        },
    });
    const tree = renderer
        .create(
            <Provider store={store}>
                <NotificationsEnabledCheckbox />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders correctly for enabled", () => {
    const scheduledTime = new Date("2029-09-06T15:00:00.000Z");
    const store = createMockStore([])({
        settings: {
            notificationsEnabled: true,
            scheduledTime,
        },
    });
    const tree = renderer
        .create(
            <Provider store={store}>
                <NotificationsEnabledCheckbox />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
