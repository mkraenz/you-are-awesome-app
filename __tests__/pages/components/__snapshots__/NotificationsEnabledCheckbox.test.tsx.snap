// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for disabled 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2289dc",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 0,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="checkbox"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "flex": 1,
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
      testID="checkboxTitle"
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Enable daily notifications
    </Text>
    <Text />
  </View>
</View>
`;

exports[`renders correctly for enabled 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2289dc",
      "borderColor": "#ededed",
      "borderRadius": 3,
      "borderWidth": 0,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="checkbox"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "flex": 1,
          "fontWeight": "bold",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
      testID="checkboxTitle"
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Enable daily notifications
Next at 5:00 PM
    </Text>
    <Text />
  </View>
</View>
`;
